static/js/settings_org.js

```diff
@@ -450,6 +450,39 @@ function _set_up() {
         var subsection = $(element).closest('.org-subsection-parent');
         return subsection.find("input[id^='id_realm_'], select[id^='id_realm_'], textarea[id^='id_realm_']");
     }
+    function changeSaveButtonState($element, state) {
+      var showHideElement = function(state) {
+        if(state === 'show') {
+          $element.removeClass('hide').addClass('show').animate({ opacity:1 }, 100);
+        } else {
+          $element.animate({opacity:0});
+        }
+      }
+      var $saveBtn = $('.save-button', $element);
+      var discardBtn = $('.discard-button', $element);
+      if(state !== 'saving') $saveBtn.removeClass('saving');
+      if(state === "unsaved") {
+        $('.icon-button-text', $saveBtn).text(i18n.t("Save Changes"));
+        $saveBtn.attr("data-status", "unsaved");
+        showHideElement('show');
+      } else if(state == "saved") {
+        $('.icon-button-text', $saveBtn).text(i18n.t("Save Changes"));
+        $saveBtn.attr("data-status", "");
+        showHideElement('hide');
+      } else if(state == "discarded") {
+        $element.removeClass('saving');
+        showHideElement('hide');
+      } else if(state == "saving") {
+        $saveBtn.addClass('saving');
+        $('.icon-button-text', $saveBtn).text(i18n.t("Saving"));
+        $saveBtn.attr("data-status", "saving");
+      } else if(state == "failed") {
+        $('.icon-button-text', $saveBtn).text(i18n.t("Save Changes"));
+        $saveBtn.attr("data-status", "failed");
+      } else if(state === 'succeeded') {
+        showHideElement('hide');
+      }
+    }
```

Comment:
I think you could probably split this commit into 2: first a commit that extracts this function (without changing behavior; just moving the code around), and then a second that changes things.  That might be more readable.

Reply:
Hi tim - I'm not sure if I completely understand what you mean in your comment. This is a new function written to put all save-discard button state changes into one function that could ( and likely should at some point be extracted out into its own file ). When you say move code around I don't completely follow.